pkg/controller/elasticsearch/user/user_provided.go

```diff
@@ -141,6 +148,41 @@ func retrieveUserProvidedFileRealm(c k8s.Client, es esv1.Elasticsearch, recorder
 	return aggregated, nil
 }
 
+func realmFromBasicAuthSecret(secret corev1.Secret, existing filerealm.Realm) (filerealm.Realm, error) {
+	realm := filerealm.New()
+	nsn := k8s.ExtractNamespacedName(&secret)
+	username := k8s.GetSecretEntry(secret, corev1.BasicAuthUsernameKey)
+	if username == nil {
+		return realm, fmt.Errorf("username is required but was empty: %v", nsn)
+	}
+	password := k8s.GetSecretEntry(secret, corev1.BasicAuthPasswordKey)
+	if password == nil {
+		return realm, fmt.Errorf("password is required but was empty: %v", nsn)
+	}
+	const minPasswordLength = 6
+	if len(password) < minPasswordLength {
```

Comment:
Not sure if we want to go that far but Elasticsearch `username` validation function is [here](https://github.com/elastic/elasticsearch/blob/ce9819fa6cc1aed8d52a09ec28503487e25bb6f8/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/support/Validation.java#L135-L159)